internals/onepassword/onepassword.go

```diff
@@ -14,55 +13,34 @@ import (
 	"github.com/mitchellh/go-homedir"
 )
 
-func CreateVault(name string) error {
-	_, err := execOP("create", "vault", name)
-	if err != nil {
-		return fmt.Errorf("could not create vault '%s': %s", name, err)
-	}
-	return nil
+type OPCLI interface {
+	IsV2() bool
+	CreateVault(name string) error
+	CreateItem(vault string, template *ItemTemplate, title string) error
+	SetField(vault, item, field, value string) error
+	GetFields(vault, item string) (map[string]string, error)
+	ExistsVault(vaultName string) (bool, error)
+	ExistsItemInVault(vault string, itemName string) (bool, error)
 }
 
-func CreateItem(vault string, template *ItemTemplate, title string) error {
-	jsonTemplate, err := json.Marshal(template)
+func GetOPClient() (OPCLI, error) {
+	out, err := execOP("--version")
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	encodedTemplate := base64.RawURLEncoding.EncodeToString(jsonTemplate)
+	version := strings.TrimSpace(string(out))
 
-	_, err = execOP("create", "item", "apicredential", "--vault="+vault, encodedTemplate, "title="+title)
-	return err
-}
-
-func SetField(vault, item, field, value string) error {
-	_, err := execOP("edit", "item", item, fmt.Sprintf(`%s=%s`, field, value), "--vault="+vault)
-	if err != nil {
-		return fmt.Errorf("could not set field '%s'.'%s'.'%s'", vault, item, field)
+	if strings.HasPrefix(version, "2.") {
+		return &OPV2CLI{
+			version: version,
+			isV2:    true,
+		}, nil
 	}
-	return nil
-}
-
-// GetFields returns a title-to-value map of the fields from the first section of the given 1Password item.
-// The rest of the fields are ignored as the migration tool only stores information in the first
-// section of each item.
-func GetFields(vault, item string) (map[string]string, error) {
-	opItem := struct {
-		Details ItemTemplate `json:"details"`
-	}{}
-	opItemJSON, err := execOP("get", "item", item, "--vault="+vault)
-	if err != nil {
-		return nil, fmt.Errorf("could not get item '%s'.'%s' from 1Password: %s", vault, item, err)
-	}
-	err = json.Unmarshal(opItemJSON, &opItem)
-	if err != nil {
-		return nil, fmt.Errorf("unexpected format of 1Password item in `op get item` command output: %s", err)
-	}
-
-	fields := make(map[string]string, len(opItem.Details.Sections[0].Fields))
-	for _, field := range opItem.Details.Sections[0].Fields {
-		fields[field.Title] = field.Value
-	}
-	return fields, nil
+	return &OPV1CLI{
```

Comment:
Nit: I think we should check whether the version starts with "1." before defaulting to v1. It might be that there is something wrong with the `op` installation, and `--version` throws an error.

Reply:
We can add an extra verifier for v1 for sanity check.

As for your concern, if the command throws an error (even one thrown by the CLI, e.g. invalid flag) the error will be thrown before returning the CLI client. Check lines 27-30 and `execOP` for more details.